src/coredump/coredump.c

```diff
@@ -706,14 +707,16 @@ static int submit_coredump(
         _cleanup_close_ int coredump_fd = -1, coredump_node_fd = -1;
         _cleanup_free_ char *core_message = NULL, *filename = NULL, *coredump_data = NULL;
         uint64_t coredump_size = UINT64_MAX;
-        bool truncated = false;
+        bool truncated = false, journald_crash;
         int r;
 
         assert(context);
         assert(iovec);
         assert(n_iovec_allocated >= n_iovec + SUBMIT_COREDUMP_FIELDS);
         assert(input_fd >= 0);
 
+        journald_crash = streq_ptr(context[CONTEXT_UNIT], SPECIAL_JOURNALD_SERVICE);
```

Comment:
I opted no to do this change, becuase `journald_crash` is only used in this function, so there'd be no code reuse, and it's used in ternary context where a function call would be awkward. And I have to say that I kind of like that this ugly check is open-coded: at least it's clear what is happening.